gateway/src/node/client_handling/websocket/connection_handler/fresh.rs

```diff
@@ -720,111 +728,125 @@ where
     {
         trace!("Started waiting for authenticate/register request...");
 
-        while let Some(msg) = self.read_websocket_message().await {
-            let msg = match msg {
-                Ok(msg) => msg,
-                Err(source) => {
-                    debug!("failed to obtain message from websocket stream! stopping connection handler: {source}");
-                    return Err(InitialAuthenticationError::FailedToReadMessage { source });
-                }
-            };
+        let mut shutdown = self.shutdown.clone();
+        loop {
+            tokio::select! {
+                biased;
+                _ = shutdown.recv() => {
+                    trace!("received shutdown signal while performing initial authentication");
+                    return Err(InitialAuthenticationError::ReceivedShutdown);
+                },
+                msg = self.read_websocket_message() => {
```

Comment:
Nit, you might find it shorter to rewrite that with else-statement:

```rs
tokio::select! {
    biased;
    _ = shutdown.recv() => {
        // ..
    },
    Some(msg) = self.read_websocket_message() => {
        // ..
    },
    else => break,
}
```

Reply:
I like the current form a bit more because it's more explicit. In case a different select branch gets added with the match pattern, it seems to be it wouldn't be clear which one caused the else branch to get executed.